doc/manual/rule_uid_schema.md

```diff
@@ -0,0 +1,139 @@
+<!---
+This Source Code Form is subject to the terms of the Mozilla
+Public License, v. 2.0. If a copy of the MPL was not distributed
+with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+-->
+
+# Rule UID Schema
+
+## Introduction
+
+The document describes the composition of the unique identifier (UID) to be associated with quality checker rules.
+
+The document describes composition of rule UIDs, declares some of the concepts to be used for ASAM rules, and provides an example for an UID for rule of the ASAM OpenDRIVE standard.
+
+## Rule UID Concepts
+
+The UID is a string which encapsulate a sequence of **concepts** that allow to identify immediately a rule across the different domain. The concept are ordered and separated via the separation character **`:`**.
+
+The concepts for the rule UID are:
+
+ * **Emanating Entity**: a domain name for the entity (organization or company) that declares the rule UID
+ * **Reference Standard**: a short string that represents the standard or the domain to which the rule is applied
+ * **Definition Settings**: the semantic version of the standard or the domain to which the rule appears or is applied **for the first time**
+ * **Rule Full Name**: the full name of the rule, as comma separated, snake lower case string. The full name of a rule is composed by the **rule set**, a categorization for the rule, and the rule **name**, a unique string inside the categorization. The rule set can be nested (meaning that can be defined as an arbitrary sequence of comma separated names, while the name is the snake case string after the last comma of the full name)
+ * Optional **Version**: a counter value representing the current version of the rule content. If content of the rule changes without changing the underlying logic for validation, this counter increases. If the change in the rule forces a modification in the validation logic, the change shall be treated as a new rule.
+
+To provide a visual description for a rule UID:
+
+```
+# Without version
+<emanating-entity>:<reference-standard>:x.y.z:rule_set.for_rules.rule_name
+# With Version
+<emanating-entity>:<reference-standard>:x.y.z:rule_set.for_rules.rule_name:2
+```
+
+> Third party rule UID creators (i.e., emanating entities different than ASAM) should still fill all the concepts above. If that is not possible, concepts shall be left blank, but separation `:` is still required (i.e., `example.com:::rulename`is valid) 
+
+UID are designed to be queried, e.g., implementations may use UNIX pattern matching.
+
+### Full Pattern
+
+```
+^
+ (?P<ENTITY>(\w+(\.\w+)+)):
+ (?P<STANDARD>([a-z]+))?:
+ (?P<STD_VERSION>([0-9]+(\.[0-9]+)+))?:
+ (?P<RULESET>(([a-z][\w_]*)\.)*)(?P<NAME>[a-z][\w_]*)
+ (:(?P<RULE_VERSION>(\d+)))?
+$
+```
+
+### ASAM Concepts
+
+ * Emanating Entity: `asam.net`
```

Comment:
`asam.net` or just `asam`? The correct name of the org is ASAM e.V., but I don't think we need the "e.V.", neither the domain. Just my personal opinion.

Reply:
I like domains because you can `whois` them, and are registered entities. It should also avoid duplicates in general.

But this is not a strong opinion and I'm open to changes.

Reply:
Ah, understand.... `whois` is an argument for using the domain. So, ok for me.